app/models/csv_exporter.rb

```diff
@@ -0,0 +1,83 @@
+require 'csv'
+
+class UnknownExporter < StandardError; end
+
+class CsvExporter
+  def initialize(mode)
```

Comment:
Nearly every method in this class is switching based on the mode. Should it just be two classes?